d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -222,15 +222,24 @@ private void handleD2NodeResponse(DiscoveryResponseData data)
       try
       {
         XdsD2.Node d2Node = resource.getResource().unpack(XdsD2.Node.class);
```

Comment:
as we discussed in 1:1 syncs previously, XdsClientImpl needs unit tests to verify the scenarios for data removal, null-guard, and empty-data-guard.

Reply:
Yes. I will add more `unit test` for `XdsClientImpl` after this initialization review finished to avoid any gap.